Add eventual consistency considerations to Consul migration docs (#3036)
hashicorp/web-unified-docs: content/vault/v1.19.x/content/docs/concepts/integrated-storage/migration-checklist.mdx

rather than in batched snapshots as Consul does, it is important to monitor IOPS as well
as disk queues to limit storage bottlenecks.

### Consistency model

> Without performance standby nodes, Integrated Storage is strongly consistent.

Reviewer: ```suggestion
Without performance standby nodes, integrated storage is strongly consistent.
```
Edited to match previous suggestion

Regular standby nodes forward all requests to the active node, so every read

> reflects the latest committed state — the same behavior as Consul with
> `consistency_mode = "strong"`.

Reviewer: ```suggestion
reflects the latest committed state. Vault provides the same behavior when you use Consul with `consistency_mode = "strong"`.
```

If you plan to use **performance standby nodes** (Vault Enterprise), the
consistency model changes. Performance standbys serve read-only requests
locally from their own copy of the Raft log. Writes go to the active node,
which then ships updates to other nodes asynchronously. In the window between
a write landing on the active node and that update reaching a standby, the

> standby presents a stale view of the data. This means read-after-write
> consistency is not guaranteed — even across requests to the same performance

Reviewer: ```suggestion
standby presents a stale view of the data. The potential for stale data means that Vault cannot guarantee read-after-write consistency, even across requests to the same performance standby.
```

standby.

> This read-after-write consistency risk is specific to Integrated Storage with
> performance standby nodes.

Reviewer: ```suggestion
The risk of read-after-write inconsistency is specific to integrated storage with
performance standby nodes.
```

Vault provides mitigations for this behavior. Refer to [Eventual consistency in

> Vault replication](/vault/docs/enterprise/consistency) for details.

Reviewer: ```suggestion
Vault provides mitigations for read-after-write inconsistency. Refer to [Eventual consistency in
Vault replication](/vault/docs/enterprise/consistency) for details.
```
